compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/ir/DeclarationFactory.kt

```diff
@@ -9,7 +9,7 @@ import org.jetbrains.kotlin.ir.declarations.*
 import org.jetbrains.kotlin.ir.types.IrType
 
 interface DeclarationFactory {
-    object FIELD_FOR_OUTER_THIS : IrDeclarationOriginImpl("FIELD_FOR_OUTER_THIS")
+    object FIELD_FOR_OUTER_THIS : IrDeclarationOriginImpl("FIELD_FOR_OUTER_THIS", isSynthetic = true)
```

Comment:
@pyos Please check native compilation and tests

Reply:
Native backend appears to have its own object for this origin and does not reference the common DeclarationFactory at all: https://github.com/JetBrains/kotlin-native/blob/54881b421cbdc784102d0023d93528f6523984b8/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/Context.kt#L72

Reply:
Also, current kotlin-native master fails to build with kotlin master because some IrUtils stuff appears to be duplicated.

Reply:
Yeah, Native didn't use DeclarationFactory at all